src/main.rs

```diff
@@ -52,6 +55,12 @@ async fn main() -> error::Result<()> {
 
     let mut state = state::new_state(config, store)?;
 
+    // Fetch public key so it's cached for the first 6hrs
+    let public_key = state.relay_client.public_key().await;
+    if public_key.is_err() {
+        warn!("Failed initial fetch of Relay's Public Key, this may prevent webhook validation.")
+    }
```

Comment:
so echo server will still boot if relay is down? (correct behavior as relay is an upstream not downstream dependency)

Reply:
Yes, if the relay is down it will boot but warn it couldn't get the key. However webhooks can't be verified and therefore can't be delivered which is fine as the relay would do that anyway.